### PR TITLE
5 feature 5 release mariogatchi and mariogatchi funeral

### LINKED ISSUE #5: Feature 5: Release Mariogatchi and Mariogatchi funeral

### PR DESCRIPTION
**Pull request by Mapsee**
Fully implemented and tested the Removal feature for Mariogatchi
Includes RemoveMariogatchiFactory abstract factory and transferMariogatchiFactory (user able to transfer Mariogatchi), killMariogatchiFactory (events that trigger Mariogatchi death)

RemoveMariogatchiFactory is an interface for method of either transferring use cases or killing the Mariogatchi use cases 
Added comments for each file for programmer to see each implementation

See classes under killmariogatchi directory for Mariogatchi death use cases
See classes under transfermariogatchi directory for Mariogatchi "release" use cases 

See RemoveMariogatchiPresenter for fun dialogue of each removal action :) 

Implementation of MariogatchiManager to handle removing and adding from a Users list of Mariogatchis

Code tested and works alongside placeholder of Mariogatchi, Inventory, as well as coded main User class

### CODE REVIEW

**Review by nishi003 — DISMISSED** [review]
Very well-written code! Good job commenting on the uses and instance variables for the different methods! Thought I'm not sure merging Mariogatchis is a feature we've discussed.

**Comment by Mapsee** [comment]
> Very well-written code! Good job commenting on the uses and instance variables for the different methods! Thought I'm not sure merging Mariogatchis is a feature we've discussed.

Thank you for your input! Merging Mariogatchi is a potential feature that might be implemented into our game

**Review by nishi003 — DISMISSED** [review]
Good work changing the Displayer to the DisplayerInterface, it follows the structure of the majority of the other code!

**Review by Grace199 — APPROVED** [review]
Looks great.

**Review by SpecialThing44 — APPROVED** [review]
Generally looks good, check case convention on methods. Use of enum's is nice although consider trying to reuse the enum in KillMariogatchiFactory. Should make the controller code easy to implement by just specifying the cause and user, and then depending on the case either taking in the Mariogatchi as an input or just picking a random one to kill. Error handling is also nice.

File: src/main/java/mariogatchi/removemariogatchi/factories/TransferMariogatchiFactory.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package mariogatchi.removemariogatchi.factories;
+
+
+import mariogatchi.removemariogatchi.transfermariogatchi.ErrorExecution;
+import mariogatchi.removemariogatchi.transfermariogatchi.Merge;
```
Comment (vivekvzdsw) [inline src/main/java/mariogatchi/removemariogatchi/factories/TransferMariogatchiFactory.java:5]: Minor but could use a * after transfermariogatchi to save lines

**Review by vivekvzdsw — APPROVED** [review]
Looks good and follows CA. Wasn't sure if the killmariogatchi package should have been condensed at first but it makes sense how you want to create independence. Minor change but in KillMarioGatchiFactory and RemoveMariogathiFactory consider using the ENUM and when using certain imports save lines of code using * .

File: src/main/java/mariogatchi/removemariogatchi/factories/TransferMariogatchiFactory.java (hunk 1 above)
Comment (Samuel-Glasroth) [reply src/main/java/mariogatchi/removemariogatchi/factories/TransferMariogatchiFactory.java:5]: dont use import all bad practice

**Pull request merged**